filter/comp.go

```
@@ -0,0 +1,215 @@
+package filter
```

You can panic instead of returning `invalid` for unreachable code.